README.md

```diff
@@ -88,22 +98,50 @@ export interface ErrorMessages {
   callParenthesesMissing: string;
   /** This variable is not defined. */
   undefinedVariable: string;
-  /** Expected at least one parameter in a function call. */
-  expectedAtLeastOneParameter: string;
-  /** Expected that many parameters, found this many. */
-  parameterCountMismatch: (thatMany: number, thisMany: number) => string;
+  /** Expected exactly that many parameters, found this many. */
+  expectedExactArgumentCount: (count: number) => string;
+  /** Expected at least that many parameters, found this many. */
+  expectedMinimumArgumentCount: (count: number) => string;
+  /** Expected between that and that many parameters, found this many. */
+  expectedArgumentCountRange: (
+    actualCount: number,
+    minCount: number,
+    maxCount: number
+  ) => string;
 }
 ```
 
+### Custom Functions
+
+The configurations of custom functions allows defining the minimum and maximum number of arguments that can be passed to each function. When a function supports an unknown number of maximum arguments, `Infinity` should be provided. If `null` is provided as value, by default, the minimum will be `0` and the maximum `Infinity`. When a is provided, both `minArgs` and `maxArgs` are required.
+
+```ts
+/** Configurations functions  */
+export type DentakuFunctionConfig = {
+  minArgs: number;
+  maxArgs: number;
+};
+
+// Example of custom functions configuration
+const customFunctions: Record<string, DentakuFunctionConfig> = {
+  fnWithOneArg: {
+    minArgs: 1,
+    maxArgs: 1,
+  },
+  fnWithTwoOrMoreArgs: {
+    minArgs: 2,
+    maxArgs: Infinity,
+  },
+  fnWithDefaultConfig: null,
```

Comment:
issue: this would fail TypeScript under strict null safety, and generally I'm not a fan of using `null` to indicate an unbounded arity. Doesn't sound semantic enough to me.

suggestion: as I suggested above, we can make `minArgs` and `maxArgs` optional with clear defaults. Then a function of unbounded arity can be represented with an empty object `{}`. When sitting alongside other objects with keys `minArgs` and `maxArgs`, I think the meaning of an empty object is clearer than the meaning of `null`

Reply:
Totally agree. Now that the config can have optional params, that's way better.